snapcraft/storeapi/__init__.py

```diff
@@ -208,6 +242,33 @@ def _is_downloaded(self, path, expected_sha512):
         return expected_sha512 == file_sum.hexdigest()
 
 
+class SSOClient(Client):
+    """The Single Sign On server deals with authentification.
+
+    It is used directly or indirectly by other servers.
+
+    """
+    def __init__(self, conf):
+        super().__init__(conf, os.environ.get(
+            'UBUNTU_SSO_API_ROOT_URL',
+            constants.UBUNTU_SSO_API_ROOT_URL))
+
+    def get_unbound_discharge(self, email, password, one_time_password,
+                              caveat_id):
+        data = dict(email=email, password=password,
+                    caveat_id=caveat_id)
+        if one_time_password:
+            data['otp'] = one_time_password
+        response = self.post(
+            'tokens/discharge', data=json.dumps(data),
+            headers={'Content-Type': 'application/json',
+                     'Accept': 'application/json'})
+        if response.ok:
+            return response.json()['discharge_macaroon'], None
+        else:
+            return None, response.text
```

Comment:
@facundobatista what about returning the errors in the response? Should that be an exception too?
Like:
if error:
    result['errors'] = error

Some of these statements come from pindonga's first implementation.